ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceDownloadManager.swift

```diff
@@ -125,14 +133,15 @@ public class ResourceDownloadManager {
   public func downloadKeyboard(withID keyboardID: String,
                                languageID: String,
                                isUpdate: Bool,
-                               fetchRepositoryIfNeeded: Bool = true) {
+                               fetchRepositoryIfNeeded: Bool = true,
+                               completionBlock: CompletionHandler<InstallableKeyboard, KeyboardKeymanPackage>? = nil) {
```

Comment:
This is one of the 'entrypoints' for resource downloading.  By allowing definition of a `completionBlock` here (and fully supporting it down the line), the method's caller can then locally specify how the downloaded package will be installed.

For this particular case, it's obvious that we already know what language to install for the keyboard.  That said, what's less obvious is what the app should do based on this.
- Some actual Keyman app use cases: 
    - reloading the keyboard if the package upgrades the active keyboard
    - automatically setting the installed keyboard as active
    - loading newly-downloaded lexical models if they match the active keyboard
    - changing the preferred lexical model for a language (if on the subview intended for it)

Both those use cases currently rely on buried notification handlers; it'd be quite nice to make these behaviors more explicit, since they directly affect Keyman's core state functionality.  (On the other hand, UI trimmings such as visual notification popups will be left as-is.)

Those behaviors also have somewhat nasty potential to introduce unwanted "cross-effects" during unit testing; being able to 'turn them off' for unit tests would be quite helpful, too.